vyper/semantics/analysis/local.py

```diff
@@ -495,11 +495,7 @@ def __init__(self, fn_node: Optional[ContractFunctionT] = None):
         self.func = fn_node
 
     def visit(self, node, typ):
-        if (
-            not isinstance(typ, TYPE_T)
-            and not isinstance(node, (vy_ast.Index, vy_ast.Tuple))  # can be deferred
-            and not isinstance(node.get_ancestor(), (vy_ast.Expr, vy_ast.Log))
-        ):
```

Comment:
these are the exceptions that we need to work around, or alternatively it can also be skipped by wrapping the type in a `TYPE_T` for `Expr`, `Log` and `Index` nodes.